packages/gatsby-plugin-mdx/gatsby/source-nodes.js

```diff
@@ -133,6 +135,34 @@ module.exports = (
       rawBody: { type: `String!` },
       fileAbsolutePath: { type: `String!` },
       frontmatter: { type: `MdxFrontmatter` },
+      slug: {
+        type: `String`,
+        async resolve(mdxNode, args, context) {
+          try {
+            const nodeWithContext = context.nodeModel.findRootNodeAncestor(
+              mdxNode,
+              node => node.internal && node.internal.type === `File`
```

Comment:
You can use optional chaining `node.internal?.type === \`File\``

Reply:
My fave and I missed a chance to use it?! Don't have to tell me twice :D

Reply:
Ooh interesting. I actually don't think this plugin has the babel stuff to support optional chaining yet. Could add/update but seems beyond the scope of this PR. Thoughts?